### PR TITLE
Add Where function to tf.raw_ops

### LINKED ISSUE #14778: where

### PR DESCRIPTION
**Pull request by paulaehab**
Close #14778 
I found Where function implemented in general_functions.py, so I add the alias of it in the RawOps

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 1]
```diff
@@ -3859,3 +3859,33 @@ def test_tensorflow_Zeta(
...
         q=x[1],
     )
+
+
+# Where
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Where",
+    dtype_and_input=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer"),
+        num_arrays=1,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3869]: The default value for this paramters is 1, we can skip it and use the implicit value to make the test shorter.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 2]
```diff
@@ -3859,3 +3859,33 @@ def test_tensorflow_Zeta(
         x=x[0],
         q=x[1],
     )
+
+
+# Where
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Where",
+    dtype_and_input=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer"),
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3868]: The original functions seems to support all dtypes, ideally we should use the default value for `helpers.get_dtypes` which is `valid`

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py (hunk 2 above)
Comment (paulaehab) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3868]: okay good point

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 3]
```diff
@@ -3859,3 +3859,33 @@ def test_tensorflow_Zeta(
...
+)
+def test_tensorflow_Where(
+    *,
+    dtype_and_input,
+    frontend,
+    fn_tree,
+    test_flags,
+    on_device,
+):
+    input_dtype, [condition] = dtype_and_input
```
[the remark this answers is not in the document]
Comment (paulaehab) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py]: @CatB1t I do this as ther the test of Tensorflow original function and if I tried to get condition[0] it give me an error

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 4]
```diff
@@ -3859,3 +3859,32 @@ def test_tensorflow_Zeta(
...
         q=x[1],
     )
+
+
+# Where
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Where",
+    dtype_and_input=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        num_arrays=1,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3869]: The default value still explicitly used.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py [hunk 5]
```diff
@@ -3859,3 +3859,32 @@ def test_tensorflow_Zeta(
...
+    on_device,
+):
+    input_dtype, condition = dtype_and_input
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        condition=condition,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3889]: Ideally, we should pass `condition[0]`, can you eloborate more on the error you're getting?

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py (hunk 4 above)
Comment (paulaehab) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3869]: Done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py (hunk 5 above)
Comment (paulaehab) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py:3889]: Done I solved the error it was because float16 not supported by tensorflow

**Comment by paulaehab** [comment]
@CatB1t the tests must be for numeric data only not all valid as boolean not supported in another frameworks also there is a problem in testing right now in PaddlePaddle framework only.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**